src/Workspaces/Core/Portable/Shared/Extensions/SourceTextExtensions.cs

```diff
@@ -217,7 +217,7 @@ private static void WriteChunksTo(SourceText sourceText, ObjectWriter writer, in
             // where it's partially full, we pass in a span to the section that is filled.  This will fast path on
             // netcore, though will incur a copy to pooled memory on netfx.
             if (count == buffer.Length)
-                writer.WriteValue(buffer);
+                writer.WriteCharArray(buffer);
```

Comment:
>WriteCharArray [](http://example.com/codeflow?start=23&length=14)

Oh, you're soo close to having this code not need to write out the length for each chunk (or the number of chunks). Eg, if there were a Write(CharArray/Span)Entries and corresponding read methods, then this code could be made nicer. Not required, but since we're making all this so much cleaner, it would match.

Reply:
can i leave to a followup? :)